test_ros2trace/test/test_ros2trace/test_trace.py

```diff
@@ -199,37 +203,52 @@ def run_trace_command_start(
         if wait_for_start:
             assert process.stdout
             stdout = ''
-            while 'press enter to stop...' not in stdout:
+            time_start = time.time()
+            time_now = time_start
+
+            def timedout(now: float) -> bool:
+                return timeout is not None and now - time_start > timeout
+
+            while 'press enter to stop...' not in stdout and not timedout(time_now):
                 stdout += process.stdout.read(1)
+                time_now = time.time()
+            self.assertFalse(
+                timedout(time_now),
```

Comment:
```suggestion
                timedout(),
```

Reply:
This could result in a false timeout if we exit from the `while` before it times out, but call `self.assertFalse(timedout(), ...)` after it times out.

The way I wrote it, that can't happen, since we set `time_out` before checking `timedout(timed_now)` for the `while` and then don't re-set it before calling `self.assertFalse(timedout(time_now), ...)`.

Reply:
@fujitatomoya does that make sense to you?

Reply:
ah yeah, last `time_now` needs to be cached from above. that makes sense 👍